experimental/addons/rabbitmq-operator/README.md

```diff
@@ -0,0 +1,80 @@
+# rabbitmq-operator
+
+Kubernetes operator to deploy and manage RabbitMQ clusters for more https://www.rabbitmq.com/.
+
+## Install
+
+Add experimental registry
+```
+vela addon registry add experimental --type=helm --endpoint=https://addons.kubevela.net/experimental/
+```
+
+Enable this addon
+```
+vela addon enable rabbitmq-operator
+```
+
+```shell
+$ vela ls -A | grep rabbitmq
+vela-system     addon-rabbitmq-operator         ns-rabbitmq-operator                    k8s-objects                                             running         healthy
+vela-system     └─                              rabbitmq-operator                       helm                                                    running         healthy Fetch repository successfully, Create helm release
+```
+
+Disable this addon
+```
+vela addon disable rabbitmq-operator
+```
+
+## Use
+## rabbitmq-operator
+
+After you enable this addon, create a namespace `prod`:
+
+```shell
+$ kubectl create namespace prod
+```
+
+Then apply this Application yaml to create a rabbitmq cluster:
+
+```yaml
+apiVersion: core.oam.dev/v1beta1
+kind: Application
+metadata:
+  name: rabbitmq-operator-sample
+spec:
+  components:
+    - type: "rabbitmq-operator"
+      name: "rabbitmq"
+      properties:
+        replicas: 2
+```
+
+```shell
+$ kubectl get po  -n prod  -o wide
+NAME                READY   STATUS    RESTARTS      AGE   IP            NODE       NOMINATED NODE   READINESS GATES
+rabbitmq-server-0   1/1     Running   0          2m55s   10.244.0.173   minikube   <none>           <none>
+rabbitmq-server-1   1/1     Running   0          2m55s   10.244.0.174   minikube   <none>           <none>
+```
+
+## Run the operator locally
+
+To connect with management console of Rabbitmq follow the below steps:
+
+- First, Have username and password default set by rabbitmq.
+
+```shell
+# For Username
+kubectl get secret rabbitmq-default-user -n prod --template={{.data.username}} | base64 --decode
+# For Password
+kubectl get secret rabbitmq-default-user -n prod --template={{.data.password}} | base64 --decode
+```
+
+- Port forward the service of Rabbitmq to login
+
+```shell
+kubectl port-forward service/rabbitmq -n prod 15672:15672
+```
+
+Now, Go on http://127.0.0.1:15672/ to login in management console.
+
+For documentation, dubugging or configuration visit https://www.rabbitmq.com/. 
```

Comment:
I checked the doc, and found there's still some gaps between users from installing this addon and really know how to use it in the real world.

In my view, we shouldn't use the rabbitmq and relies on the port-forward command.

A K8s native way should be:

1. get the service endpoint of the rabbitmq.
2. build a producer/consumer container images from rabbitmq tutorial, that could be the real world application using the endpoint, inject the service endpoint from step 1 into the container as ENV or volume.
3. deploy both the producer and consumer, check it works.

Then, the addon could be really adopted and understood well.


Hope we can make it more useful in this way, thanks @MdSahil-oss

Reply:
We also have a `vela status --endpoint` command that could get service automatically, even for CRD which has the   [resource relationship](https://kubevela.net/docs/reference/topology-rule) decalred.